### PR TITLE
Safety check before clean

### PR DESCRIPTION
**Pull request by ganwell**
Check the Makefile in the build directory for a token to make sure the
$(BUILD) variable point to the correct directory.

### CODE REVIEW

**Comment by ganwell** [comment]
@winged I don't get it. The values are hard-coded, they are never going to change. How can this not be future proof? If anybody tempers with any of the variables he can't clean anymore and therefore will fix the problem again. If anybody removes the check, it will be the same with any other typo of check. What has to be cleaned is never going to change.

Please explain.

**Comment by ganwell** [comment]
@winged Ok, I made it a bit less future-proof, but instead a bit more restrictive. I guess even without the checks it is now safe-enough, but I like the checks very much, they can't be accidentally cheated.

**Comment by ganwell** [comment]
@winged Now only if one starts a new directory, the clean target has to be adjusted, new files in existing directory are ok.

File: mk/base.mk [hunk 1]
```diff
@@ -209,7 +209,14 @@ uninstall:  ## Uninstall chirp
 	rm -rf $(DEST)$(PREFIX)/share/doc/chirp
 
 clean:  # Clean chirp
+	@cd "$(BUILD)" && grep -q cf65e84fdbb7644a0c7725ebe6259490 Makefile
```
Comment (winged) [inline mk/base.mk]: Can you add a comment to explain why this is done? Then I think my concern regarding some future maintainer can be considered "closed" :)

**Pull request merged**